--- junit-original.py	2022-08-26 13:20:23.148507399 +1000
+++ junit.py	2022-08-26 13:20:47.087450150 +1000
@@ -108,7 +108,7 @@
             ElementTree.ElementTree._write(self, file, node, encoding,
                                            namespaces)
 
-if hasattr(ElementTree, '_serialize'):
+if hasattr(ElementTree, '_serialize') and False:
     # pylint: disable=protected-access
     def _serialize_xml2(write, elem, encoding, qnames, namespaces,
                         orig=ElementTree._serialize_xml):
