diff --git a/build.vc15/lib_mpfr/lib_mpfr.vcxproj b/build.vc15/lib_mpfr/lib_mpfr.vcxproj
index a3c72e48..85da0bcc 100644
--- a/build.vc15/lib_mpfr/lib_mpfr.vcxproj
+++ b/build.vc15/lib_mpfr/lib_mpfr.vcxproj
@@ -148 +148 @@
-      <RuntimeLibrary>MultiThreaded</RuntimeLibrary>
+      <RuntimeLibrary>MultiThreadedDLL</RuntimeLibrary>
@@ -179 +179 @@
-      <RuntimeLibrary>MultiThreaded</RuntimeLibrary>
+      <RuntimeLibrary>MultiThreadedDLL</RuntimeLibrary>
