diff --git a/common/libBuffer/include/CommonFWriteBufferFlusher.h b/common/libBuffer/include/CommonFWriteBufferFlusher.h
index c7af45b2..62b0f991 100644
--- a/common/libBuffer/include/CommonFWriteBufferFlusher.h
+++ b/common/libBuffer/include/CommonFWriteBufferFlusher.h
@@ -58,7 +58,7 @@ namespace Common
 #else
 		typedef __int64 FilePosType;
 #endif
-	typedef std::tr1::unordered_map<MarkId, FilePosType > MarkIdToFilePos;
+	typedef std::unordered_map<MarkId, FilePosType > MarkIdToFilePos;
 
 	public:
 		static const size_t DEFAUL_BUFFER_SIZE = 64*1024;
